src/pybind/ceph_volume_client.py

```diff
@@ -1442,6 +1451,34 @@ def get_object(self, pool_name, object_name):
             ioctx.close()
         return bytes_read
 
+    def update_object(self, pool_name, object_name, data_modifier):
+        iterations = 0
+
+        while True:
+            data, version1 = self.get_object_and_version(pool_name, object_name)
+            result_data = data_modifier(data)
+            version2 = self.put_object(pool_name, object_name, result_data)
+            if version2 == version1 + 1:
+                return
+
+            iterations += 1
+            if iterations > 20:
```

Comment:
nit: better to define the iteration upper bound elsewhere (maybe a class variable in the exception class)?

Reply:
If we keep iterations,  IMO, it would be nice to do so.